gdk/gdkglcontext.c

```diff
@@ -275,13 +341,14 @@ gdk_gl_context_download_texture (GdkGLContext    *context,
       if (priv->use_es)
         {
           for (i = y; i < height; i++)
-            glReadPixels (x, i, width, 1, GL_BGRA, GL_UNSIGNED_INT_8_8_8_8_REV,
-                          (unsigned char *) data + (i * stride));
+            glReadPixels (x, i, width, 1, GL_RGBA, GL_UNSIGNED_BYTE,
+                         (unsigned char *) data + (i * stride));
+            image_surface_rgba_to_argb (image_surface);
```

Comment:
I'd probably just add a:

```
if (priv->use_es)
  image_surface_rgba_to_argb (image_surface)
```

at the end of the function, since it's something we want to do in both cases.